main/samba/samba.initd

```diff
@@ -1,24 +1,41 @@
 #!/sbin/openrc-run
 
 extra_started_commands="reload"
+piddir=${piddir:-"/run/samba"}
+daemon_list="${daemon_list//samba/}"
 
 DAEMON=${SVCNAME#samba.}
 if [ "$DAEMON" != "samba" ]; then
 	daemon_list=$DAEMON
+elif [ "${daemon_list// /}" == "" ] || [ -z '$daemon_list' ]; then
+	daemon_list="samba"
```

Comment:
Your 2 daemon_list changes above will make it impossible only start `nmbd` and `smbd` if you set  `daemon_list="smbd nmbd samba"` (eg. it will not start `samba` service).

I would prefer that `/etc/init.d/samba` would start all services specified in `daemon_list` (including `samba` if its specified), while `/etc/init.d/samba.<daemon>` only starts `<daemon>`, even if `<daemon>` is `samba`.

Reply:
When daemon_list="smbd nmbd samba" samba discarded right at start because daemon_list="${daemon_list//samba/}". We don't want mix them.

I will think about a POSIX friendly solution. Thanks for review. You can hold for this PR for later  because I can do a version upgrade as well because samba is now 4.8.5.  You may do a simple rebuild of samba against #5045 because ldb is more demanding right now.

Reply:
It seems PR already merged. Thanks.